src/coreclr/jit/llvm.cpp

```diff
@@ -1620,12 +1620,23 @@ void Llvm::buildReturn(GenTree* node)
         case TYP_UINT:
         case TYP_LONG:
         case TYP_ULONG:
+        case TYP_STRUCT:
             if (node->gtGetOp1()->TypeIs(TYP_FLOAT))
             {
                 // TODO-LLVM: remove this case by lowering see
                 // https://github.com/dotnet/runtimelab/pull/2007#issuecomment-1264715441
                 failFunctionCompilation();
             }
+            if (node->gtGetOp1()->IsIntegralConst(0))
+            {
+                // special case returning 0 initialized structs
+                Type* structLlvmType = getLlvmTypeForCorInfoType(_sigInfo.retType, _sigInfo.retTypeClass);
+                Value* structAddrValue = _builder.CreateAlloca(structLlvmType, 0U);
+                _builder.CreateMemSet(structAddrValue, _builder.getInt8(0), _builder.getInt32(structLlvmType->getScalarSizeInBits() >> 3), {});
+                _builder.CreateRet(_builder.CreateLoad(structAddrValue));
+
```

Comment:
```suggestion
                // Special-case returning zero-initialized structs.
                Type* structLlvmType = getLlvmTypeForCorInfoType(_sigInfo.retType, _sigInfo.retTypeClass);
                Value* structAddrValue = _builder.CreateAlloca(structLlvmType, 0U);
                Value* structSizeValue = _builder.getInt32(structLlvmType->getScalarSizeInBits() / TARGET_POINTER_SIZE);
                _builder.CreateMemSet(structAddrValue, _builder.getInt8(0), structSizeValue, {});
                _builder.CreateRet(_builder.CreateLoad(structAddrValue));
```

Reply:
Isn't that changing a division by 8 (going from bits to bytes) to a division by 4 (`TARGET_POINTER_SIZE` for wasm32).  If the `>> 3` is not great, we can just do `/ 8` to go from bites to bytes?

Reply:
Ah, sorry, indeed it should be `/ BITS_PER_BYTE`.